### ISSUE TITLE
CLI prints "null" and exits with error code 1

### ISSUE THREAD

**Issue by pdcastro** [balena-io/balena-cli#1333]
This error is rare but can happen with any balena CLI command, any platform (Windows, Mac, Linux) and any installation type (npm dev, npm production, standalone zip, executable installer). The most recent example was issue #1331, and it is closely related to issue #1193.

It happens because of an issue / bug in the Sentry library, which is supposed to intercept errors and report them to Balena via the [sentry.io](https://sentry.io/) platform. It seems that the Sentry library (at least the old version being used) cannot cope with certain types of errors, and then itself fails and "masks" the underlying error, resulting in "null" being printed to the console:

```
$ balena login
null
```

That's a rather frustrating error, difficult to debug. A workaround (to reveal the underlying error) is to sidestep the Sentry library by commenting out (`//`) the `setupRaven()` line in the `build/app-common.js` file, so that it looks as follows:
```
    // setupRaven();
```

If the CLI was installed with the executable (native) installers, the file is located at:

OS | File
-- | --
Windows | `C:\Program Files\balena-cli\client\build\app-common.js`
macOS | `/usr/local/lib/balena-cli/build/app-common.js`

With the "npm installation", it depends on where Node.js was installed and whether `nvm` was used, but typical locations are:

OS | File
-- | --
Linux | /usr/local/lib/node_modules/balena-cli/build/app-common.js
nvm | ~/.nvm/versions/node/vX.Y.Z/lib/node_modules/balena-cli/build/app-common.js

This workaround is not possible in the standalone zip packages, as the source code is bundled in the executable binary.

The first step towards a proper fix is probably to update the Sentry library being used by the CLI (#1523). Unfortunately it's not just a matter of updating a dependency version, because Sentry did a re-writing of the public API.

**Comment by pdcastro** [comment]
Draft pull request: https://github.com/balena-io/balena-cli/pull/1295

**Comment by pdcastro** [comment]
"hello" support conversation: https://app.frontapp.com/open/cnv_1yv269v

**Comment by pdcastro** [comment]
On macOS, in order to edit the `/usr/local/lib/balena-cli/build/app-common.js` file, you can use the `nano` command-line text editor. Open a terminal prompt and run:
```
$ sudo nano /usr/local/lib/balena-cli/build/app-common.js
Password:
```
It will prompt you for the computer login password because the the file is owned by the `root` user. The window will look something like the following screenshot. User the arrow keys to navigate towards the end of the file, and type `//` before `setupRaven();` (in the `globalInit()` function) as shown in the screenshot. Then type Ctrl-O to save the file, and Ctrl-X to exit.

![nano](https://user-images.githubusercontent.com/15091591/60987985-a1d48180-a33a-11e9-9750-df5679d1984e.png)

**Comment by alsmith** [comment]
I ran into this problem and after commenting out the setupRaven() call, it emerged that the problem was due to $HOME/.balena being owned by root. It was empty anyway so I got rid of it, uncommented setupRaven() again and everything works now.

**Comment by HAKASHUN** [comment]
Though I am using an older version of cli (resin-cli@8.1.0), I encountered the same error.

I fixed part of the `require('any-promise/register/bluebird')` code because an Error is happened in [`any-promise/loader.js`](https://github.com/kevinbeaty/any-promise/blob/c2029f9e37a15b04c303bd0ec19cf16d00ec8f3d/loader.js#L53-L55).

https://github.com/balena-io/balena-cli/blob/8dc4c0871a6c22c028521751ecb8518204fdde44/lib/app.coffee#L82-L86

```javascript
// Before
require('any-promise/register/bluebird')

// After
require('any-promise')
```

The same fixes in [`app-common.ts`](https://github.com/balena-io/balena-cli/blob/062fadfa498802a454407a8fdf53d7d8e7b7961d/lib/app-common.ts#L103) may improve it 🤔 .


|`MyEnvironment`||
|----|----|
|`OS`|Mac|
|`Node`|8.9.4|
|`Cli` |resin-cli@8.1.0|

**Comment by jacobrosenthal** [comment]
Mine was 
```
Error: Setting not found: apiUrl
```
Then when digging around I found an empty ~/.balenarc.yml file which when deleted got me working again

**Comment by pdcastro** [comment]
This issue should be fixed in CLI release 11.28.14 ([releases](https://github.com/balena-io/balena-cli/releases)). If you still come across this issue ("prints null and exits") with CLI v11.28.14 or later, please leave a comment and we can reopen this issue (or you are also welcome to create a new issue).

If you needed/wanted to get around the issue by commenting out the `setupRaven()` line (as shown in a screenshot above) with newer CLI releases (11.28.14 and later), please note that that line was replaced with `await setupSentry()` (https://github.com/balena-io/balena-cli/blob/v11.28.14/lib/app-common.ts#L245), so the commented out line in newer releases would look like:
```
export async function globalInit() {
	// await setupSentry();
	...
```
